packages/path_provider/path_provider_windows/lib/src/path_provider_windows_real.dart

```diff
@@ -144,9 +146,9 @@ class PathProviderWindows extends PathProviderPlatform {
         pathPtrPtr,
       );
 
-      if (FAILED(hr)) {
+      if (hr != S_OK) {
```

Comment:
Windows can set bits in the `HRESULT`, a successful status might not be equal to `S_OK`.

I would consider adding a `FAILED` helper whose implementation is `hr < 0`.

See:
1. https://learn.microsoft.com/en-us/windows/win32/api/winerror/nf-winerror-failed
2. https://stackoverflow.com/a/12852668

Reply:
Done, thanks for catching that.